apps/passport-server/src/services/generic-issuance/subservices/PipelineAPISubservice.ts

```diff
@@ -520,4 +538,117 @@ export class PipelineAPISubservice {
       }
     );
   }
+
+  /**
+   * todo a comment
+   */
+  public async handleGetOfflineTickets(
+    request: PodboxGetOfflineTicketsRequest
+  ): Promise<PodboxGetOfflineTicketsResponseValue> {
+    return traced(SERVICE_NAME, "handleGetOfflineTickets", async (span) => {
+      logger(LOG_TAG, "handleGetOfflineTickets", str(request));
+
+      let emailAddress;
+      try {
+        emailAddress = (
+          await this.credentialSubservice.verifyAndExpectZupassEmail(
+            request.credential
+          )
+        ).emailClaim.emailAddress;
+      } catch (_e) {
+        throw new PCDHTTPError(401, "Not authorized");
+      }
+
+      const offlineTickets: PodboxOfflineTicket[] = [];
+
+      for (const pipeline of this.pipelineSubservice.getAllPipelines()) {
+        if (!pipeline.instance) {
+          continue;
+        }
+
+        for (const capability of pipeline?.instance?.capabilities ?? []) {
+          if (isCheckinCapability(capability)) {
+            tracePipeline(pipeline.definition);
```

Comment:
each invocation of `tracePipeline` overwrites the previous invocation's metadata on the current span, so this function should be called at most once in a trace